code/modules/mob/living/carbon/carbon.dm

```diff
@@ -166,6 +166,8 @@
 
 	else if(!(I.item_flags & (NODROP | ABSTRACT)))
 		thrown_thing = I
+		if(SendSignal(COMSIG_CARBON_THROW_ITEM, thrown_thing, src) & COMPONENT_STOP_THROW)
+			return

```

Comment:
This signal should be much earlier in the proc

Reply:
It's standard for us to have these sorts of signals early on in the proc so that components which prevent it do so before it has to do other checks that are now unnecesary.

Do you even use this in this pr?

Reply:
...Huh, apparently not. I thought I did. Will add in relevant PR later.

Also, I think I'll refactor all status effects, then. Component status effects would allow for non-living status effects, too, such as making burning items a status effect, et cetera.